### ISSUE TITLE
POC Remote Terminal

### ISSUE THREAD

**Issue by creationix** [virgo-agent-toolkit/rackspace-monitoring-agent#846]
- Add pty bindings to luvit
 - Add hostinfo to initiate session
 - Spawn relay server on cloud vm (implemented in luvit)
 - Write web client to relay as standalone static page.
 - Convert web client to greasemonkey script for demo.

**Comment by kaustavha** [comment]
Why not just have a web server running on the cloud vm running a luvit web server serving the page?
<cloud vm server ip>/frontdoor 
Which website are we trying to modify instead of just using a custom webpage?

**Comment by creationix** [comment]
I estimate about a day each to get the initial technical POC running, then a week to polish it and finish phase 2 to make for a sweet demo.

**Comment by creationix** [comment]
@kaustavha For the demo, greasemonkey would make it look like we've modified the production server and feel much more real.  I split up the description to put the standalone client in phase 0.

**Comment by creationix** [comment]
For the pty, we probably want to just port https://github.com/chjj/pty.js/tree/master/src to luvit.

**Comment by kaustavha** [comment]
True, Im still wondering which website we're potentially 'integrating' this feature into. 
cloudcontrol? mycloud? intelligence?

**Comment by creationix** [comment]
@kaustavha My understanding was that it's for support people, so whatever web interface they use?  I don't know exactly what that is.

**Comment by kaustavha** [comment]
@creationix Not scavenge either
https://github.com/KateAdams/lua-pty
or
https://github.com/LuaDist/lpty
?

**Comment by creationix** [comment]
Right, porting the lua bindings to luvit is actually more work than porting node bindings since translating between binding APIs is usually 1-1 mapping, but translating between event frameworks often required complete re-architectures.

**Comment by kaustavha** [comment]
Makes sense. Interesting situation.
